### PR TITLE
Added quests.json and quests_vendor.json for quest rewards

### PR DESCRIPTION
**Pull request by lherron**
Hi, I've added extracts for quest rewards by quest plus class and by vendor.

### CODE REVIEW

**Comment by brather1ng** [comment]
Thanks for the PR!

Please add docs files (and entries in the readme) for your new files. Documenting how the game data works in addition to providing it is a major part of this repo, imo.

Some other things I noticed from taking a quick look:

- If a value is already present as a key, it doesn't need to be repeated in its object (e.g. the quest ids). That's just redundant information.
- Since the code for the two files is pretty much independent, I'd prefer two separate Python files.
- Please remove commented out code.
- I don't like "All" as a class value. It's a hardcoded string with a special meaning every user has to implement. I'm not sure what the best solution would be here. What about skipping the classes entry instead?

**Comment by lherron** [comment]
Finally had some free time to make the updates, let me know if this looks good.  Thanks!

File: RePoE/parser/modules/quest_rewards.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                        "rewards": {}
+                    }
+                
+                ##### This is for root->quest->rewards->class
+                # Add an entry for the reward into the quest node
+                if rewardName not in root[questId]["rewards"]:
+                    root[questId]["rewards"][rewardId] = {
+                        "classes": [],
+                        "name": rewardName,
+                        "type": reward_row["BaseItemTypesKey"]["ItemClassesKey"]["Name"]
```
Comment (brather1ng) [inline RePoE/parser/modules/quest_rewards.py]: I'd prefer the item class id over this so it's a direct reference to `item_classes.json`.

File: RePoE/parser/modules/quest_rewards.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                    }
+                if charClass == "All":
+                    root[questId]["rewards"][rewardId]["classes"] = all_classes
+                else:
+                    if charClass not in root[questId]["rewards"][rewardId]["classes"]:
+                        root[questId]["rewards"][rewardId]["classes"].append(charClass)
+
+
+        for state_row in relational_reader['QuestStates.dat']:
```
Comment (brather1ng) [inline RePoE/parser/modules/quest_rewards.py]: With vendor rewards in a separate file, this loop does nothing here.

File: RePoE/parser/modules/vendor_rewards.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                if ((npcName != "Lilly Roth") and (npcName != "Siosa")):
+                    if reward_row["QuestState"] in all_quest_states:
+                        root[npcId]["rewards"][rewardId]["quest_id"] = all_quest_states[reward_row["QuestState"]]
+                    else:
+                        # BLATANT KLUDGE: Quest state 244 = a2q6, but isn't in QuestStates.dat
+                        if reward_row["QuestState"] == 244:
+                            root[npcId]["rewards"][rewardId]["quest_id"] = "a2q6"
+                print("Length of root: ", len(root))
+                print("Length of root[", npcId, "][rewards]: ", len(root[npcId]["rewards"]))
+        print(root)
```
Comment (brather1ng) [inline RePoE/parser/modules/vendor_rewards.py]: This looks like output from your debugging and is a bit much for general logging. Probably doesn't hurt to remove all the prints from this file.

**Review by brather1ng — CHANGES_REQUESTED** [review]
Looks mostly good now, just a few minor things.

**Comment by lherron** [comment]
Done!  Not sure how I missed those prints. ;-)

**Review by brather1ng — APPROVED** [review]

**Pull request merged**